pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -468,3 +469,14 @@ def hpc_save_path(folderpath: _PATH) -> str:
         ckpt_number = (max_suffix if max_suffix is not None else 0) + 1
         filepath = os.path.join(folderpath, f"hpc_ckpt_{ckpt_number}.ckpt")
         return filepath
+
+
+@runtime_checkable
+class _SupportsStateDict(Protocol):
```

Comment:
Not sure how types are organised in this repo, but this type should be defined in `pl.utilities.types`?

Reply:
this isn't actually a type, used to check `isinstance`. I think since it's somewhat related to checkpointing API, so added it here and @carmocca suggested it too.